xbmc/dialogs/GUIDialogBusy.cpp

```diff
@@ -79,30 +76,30 @@ bool CGUIDialogBusy::WaitOnEvent(CEvent &event, unsigned int displaytime /* = 10
   bool cancelled = false;
   if (!event.Wait(std::chrono::milliseconds(displaytime)))
   {
-    // throw up the progress
-    CGUIDialogBusy* dialog = CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogBusy>(WINDOW_DIALOG_BUSY);
+    CGUIDialogBusy* dialog = static_cast<CGUIDialogBusy*>(
```

Comment:
Since the windowmanager creates and adds this window on startup and only a single instance exists throughout the whole app lifecycle, wouldn't it make more sense for this window to be a singleton? It just looks odd that this class defines the windowmanager as a friend class and then calls the window manager to get itself

Reply:
> I thought it is more clever/safe to make the ctor/dtor private and add window manager as friend. If this is not acceptable, I could make the ctor/dtor public again.

No, maybe i did not express myself correctly. This is fine.
I know the call to `WindowManager` to get the busydialog was already there and it exists because the method is static. 
I was trying to say that the simple fact this call exists is a sign the design is a bit fragile. I mean, it doesn't make much sense to be the dialog asking the windowmanager (responsible for its creation) to get a pointer to itself so that some logic take place. Maybe this is something that can be improved later but I also don't know how to improve this at this point.

Reply:
Then let’s keep it as it is for now. Code quality did not decrease, so it should be okay.